Documentation/building/dev-workflow.md

```diff
@@ -0,0 +1,46 @@
+Dev Workflow
+===============
+
+The dev workflow describes the development process to follow when working in our repo. 
+It is divided into specific tasks that are fast, transparent and easy to understand.
+
+This provides flexibility as with discrete tasks people can iterate on a specific one without needing to run the full workflow, other processes can reuse the tasks as needed and anyone can put together the tasks as wanted to create a customized workflow.
+
+## Process 
+![Dev Workflow process](../images/dev-workflow.png)
+
+All the tasks (except enlist, code, submit) will have scripts located at the root of the repo, and will have usage help to better understand what they provide.
+In addition, we want to have an “easy button” script for running many or all of these tasks sequentially in a single command.
+
+## Tasks
+
+**Clean**
+
+Cleans the local enlistment/environment, for example removing files not in the repository. 
+
+**Sync**
+
+The sync task gets the latest source history and packages in order to eliminate all the network traffic when we build. This way we are able to hit the network only when we are intentional about doing it and then be able to build even in offline mode. 
+It is important to note that Sync gets the source history by doing `git fetch --all -p –v`.
+
+It does not update branches or merge in changes to local files, giving devs flexibility in managing their Git repos. Devs will need to manually update their local files before restoring packages.
```

Comment:
I think the idea is you use your normal git commands to get your current working directory into the state you want and then you run sync.cmd which will restore the packages. 

I think people people keep getting hung up on the fact that sync.cmd also does the git fetch and I agree that for a lot of scenarios it isn't that useful but it is still useful for the scenario where I want to get all the things from the internet and then unplug. Of course if people's current working directory is in a state that doesn't match what you ultimately want to work on offline it will not help so you may need to run sync multiple times if you plan to work in multiple branches that have different package dependencies while in an offline state.